//
// DO NOT MODIFY THIS FILE DIRECTLY!
// IT IS GENERATED BY generate_policy_source.py
// FROM ../../authpolicy/policy/resources/policy_templates.json
//


syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package enterprise_management;

message StringList {
  repeated string entries = 1;
}

message PolicyOptions {
  enum PolicyMode {
    // The given settings are applied regardless of user choice.
    MANDATORY = 0;
    // The user may choose to override the given settings.
    RECOMMENDED = 1;
    // No policy value is present and the policy should be ignored.
    UNSET = 2;
  }
  optional PolicyMode mode = 1 [default = MANDATORY];
}

message BooleanPolicyProto {
  optional PolicyOptions policy_options = 1;
  optional bool value = 2;
}

message IntegerPolicyProto {
  optional PolicyOptions policy_options = 1;
  optional int64 value = 2;
}

message StringPolicyProto {
  optional PolicyOptions policy_options = 1;
  optional string value = 2;
}

message StringListPolicyProto {
  optional PolicyOptions policy_options = 1;
  optional StringList value = 2;
}

message CloudPolicySettings {
  optional StringPolicyProto HomepageLocation = 3;
  optional BooleanPolicyProto HomepageIsNewTabPage = 4;
  optional BooleanPolicyProto AlternateErrorPagesEnabled = 7;
  optional BooleanPolicyProto SearchSuggestEnabled = 8;
  optional IntegerPolicyProto NetworkPredictionOptions = 275;
  optional BooleanPolicyProto WPADQuickCheckEnabled = 263;
  optional StringListPolicyProto DisabledSchemes = 87;
  optional BooleanPolicyProto Http09OnNonDefaultPortsEnabled = 347;
  optional BooleanPolicyProto JavascriptEnabled = 11;
  optional BooleanPolicyProto IncognitoEnabled = 12;
  optional IntegerPolicyProto IncognitoModeAvailability = 95;
  optional BooleanPolicyProto SavingBrowserHistoryDisabled = 13;
  optional BooleanPolicyProto AllowDinosaurEasterEgg = 311;
  optional StringPolicyProto RemoteAccessHostClientDomain = 318;
  optional BooleanPolicyProto RemoteAccessHostFirewallTraversal = 97;
  optional StringPolicyProto RemoteAccessHostDomain = 156;
  optional StringPolicyProto RemoteAccessHostTalkGadgetPrefix = 158;
  optional BooleanPolicyProto RemoteAccessHostRequireCurtain = 159;
  optional BooleanPolicyProto RemoteAccessHostAllowClientPairing = 236;
  optional BooleanPolicyProto RemoteAccessHostAllowGnubbyAuth = 259;
  optional BooleanPolicyProto RemoteAccessHostAllowRelayedConnection = 265;
  optional StringPolicyProto RemoteAccessHostUdpPortRange = 266;
  optional BooleanPolicyProto RemoteAccessHostMatchUsername = 287;
  optional StringPolicyProto RemoteAccessHostTokenUrl = 288;
  optional StringPolicyProto RemoteAccessHostTokenValidationUrl = 289;
  optional StringPolicyProto RemoteAccessHostTokenValidationCertificateIssuer = 290;
  optional BooleanPolicyProto PrintingEnabled = 14;
  optional BooleanPolicyProto ForceSafeSearch = 164;
  optional BooleanPolicyProto ForceGoogleSafeSearch = 284;
  optional BooleanPolicyProto ForceYouTubeSafetyMode = 285;
  optional IntegerPolicyProto ForceYouTubeRestrict = 350;
  optional BooleanPolicyProto SafeBrowsingEnabled = 16;
  optional BooleanPolicyProto PasswordManagerEnabled = 18;
  optional BooleanPolicyProto AutoFillEnabled = 20;
  optional StringListPolicyProto DisabledPlugins = 21;
  optional StringListPolicyProto EnabledPlugins = 80;
  optional StringListPolicyProto DisabledPluginsExceptions = 81;
  optional BooleanPolicyProto DisablePluginFinder = 68;
  optional BooleanPolicyProto SyncDisabled = 22;
  optional StringPolicyProto DownloadDirectory = 66;
  optional BooleanPolicyProto CaptivePortalAuthenticationIgnoresProxy = 297;
  optional StringPolicyProto ProxyMode = 23;
  optional IntegerPolicyProto ProxyServerMode = 24;
  optional StringPolicyProto ProxyServer = 25;
  optional StringPolicyProto ProxyPacUrl = 26;
  optional StringPolicyProto ProxyBypassList = 27;
  optional StringPolicyProto ProxySettings = 118;
  optional StringListPolicyProto ExtensionInstallBlacklist = 34;
  optional StringListPolicyProto ExtensionInstallWhitelist = 35;
  optional StringListPolicyProto ExtensionInstallForcelist = 36;
  optional StringListPolicyProto ExtensionInstallSources = 150;
  optional StringListPolicyProto ExtensionAllowedTypes = 170;
  optional StringPolicyProto ExtensionSettings = 280;
  optional BooleanPolicyProto ShowHomeButton = 37;
  optional BooleanPolicyProto DeveloperToolsDisabled = 38;
  optional IntegerPolicyProto RestoreOnStartup = 39;
  optional StringListPolicyProto RestoreOnStartupURLs = 40;
  optional BooleanPolicyProto BlockThirdPartyCookies = 41;
  optional BooleanPolicyProto DefaultSearchProviderEnabled = 42;
  optional StringPolicyProto DefaultSearchProviderName = 43;
  optional StringPolicyProto DefaultSearchProviderKeyword = 44;
  optional StringPolicyProto DefaultSearchProviderSearchURL = 45;
  optional StringPolicyProto DefaultSearchProviderSuggestURL = 46;
  optional StringPolicyProto DefaultSearchProviderInstantURL = 47;
  optional StringPolicyProto DefaultSearchProviderIconURL = 48;
  optional StringListPolicyProto DefaultSearchProviderEncodings = 49;
  optional StringListPolicyProto DefaultSearchProviderAlternateURLs = 163;
  optional StringPolicyProto DefaultSearchProviderSearchTermsReplacementKey = 173;
  optional StringPolicyProto DefaultSearchProviderImageURL = 231;
  optional StringPolicyProto DefaultSearchProviderNewTabURL = 239;
  optional StringPolicyProto DefaultSearchProviderSearchURLPostParams = 232;
  optional StringPolicyProto DefaultSearchProviderSuggestURLPostParams = 233;
  optional StringPolicyProto DefaultSearchProviderInstantURLPostParams = 234;
  optional StringPolicyProto DefaultSearchProviderImageURLPostParams = 235;
  optional IntegerPolicyProto DefaultCookiesSetting = 50;
  optional IntegerPolicyProto DefaultImagesSetting = 51;
  optional IntegerPolicyProto DefaultJavaScriptSetting = 52;
  optional IntegerPolicyProto DefaultPluginsSetting = 53;
  optional IntegerPolicyProto DefaultPopupsSetting = 54;
  optional IntegerPolicyProto DefaultNotificationsSetting = 55;
  optional IntegerPolicyProto DefaultGeolocationSetting = 56;
  optional IntegerPolicyProto DefaultMediaStreamSetting = 151;
  optional IntegerPolicyProto DefaultWebBluetoothGuardSetting = 322;
  optional IntegerPolicyProto DefaultKeygenSetting = 315;
  optional StringListPolicyProto AutoSelectCertificateForUrls = 104;
  optional StringListPolicyProto CookiesAllowedForUrls = 79;
  optional StringListPolicyProto CookiesBlockedForUrls = 69;
  optional StringListPolicyProto CookiesSessionOnlyForUrls = 70;
  optional StringListPolicyProto ImagesAllowedForUrls = 71;
  optional StringListPolicyProto ImagesBlockedForUrls = 72;
  optional StringListPolicyProto JavaScriptAllowedForUrls = 73;
  optional StringListPolicyProto JavaScriptBlockedForUrls = 74;
  optional StringListPolicyProto KeygenAllowedForUrls = 316;
  optional StringListPolicyProto KeygenBlockedForUrls = 317;
  optional StringListPolicyProto PluginsAllowedForUrls = 75;
  optional StringListPolicyProto PluginsBlockedForUrls = 76;
  optional StringListPolicyProto PopupsAllowedForUrls = 77;
  optional StringPolicyProto RegisteredProtocolHandlers = 270;
  optional StringListPolicyProto PopupsBlockedForUrls = 78;
  optional StringListPolicyProto NotificationsAllowedForUrls = 107;
  optional StringListPolicyProto NotificationsBlockedForUrls = 108;
  optional BooleanPolicyProto Disable3DAPIs = 57;
  optional IntegerPolicyProto PolicyRefreshRate = 58;
  optional IntegerPolicyProto MaxInvalidationFetchDelay = 230;
  optional BooleanPolicyProto ChromeOsLockOnIdleSuspend = 63;
  optional StringPolicyProto ChromeOsMultiProfileUserBehavior = 246;
  optional BooleanPolicyProto TranslateEnabled = 82;
  optional BooleanPolicyProto AllowOutdatedPlugins = 83;
  optional BooleanPolicyProto AlwaysAuthorizePlugins = 88;
  optional BooleanPolicyProto BookmarkBarEnabled = 84;
  optional BooleanPolicyProto EditBookmarksEnabled = 85;
  optional StringListPolicyProto URLBlacklist = 105;
  optional StringListPolicyProto URLWhitelist = 106;
  optional StringPolicyProto OpenNetworkConfiguration = 109;
  optional StringPolicyProto DefaultPrinterSelection = 310;
  optional BooleanPolicyProto EnableOnlineRevocationChecks = 131;
  optional BooleanPolicyProto RequireOnlineRevocationChecksForLocalAnchors = 237;
  optional BooleanPolicyProto EnableSha1ForLocalAnchors = 342;
  optional IntegerPolicyProto SAMLOfflineSigninTimeLimit = 256;
  optional BooleanPolicyProto DriveDisabled = 141;
  optional BooleanPolicyProto DriveDisabledOverCellular = 142;
  optional StringListPolicyProto PinnedLauncherApps = 146;
  optional BooleanPolicyProto DisableSafeBrowsingProceedAnyway = 152;
  optional BooleanPolicyProto SafeBrowsingExtendedReportingOptInAllowed = 301;
  optional BooleanPolicyProto SpellCheckServiceEnabled = 153;
  optional BooleanPolicyProto ExternalStorageDisabled = 154;
  optional BooleanPolicyProto ExternalStorageReadOnly = 345;
  optional BooleanPolicyProto AudioOutputAllowed = 161;
  optional BooleanPolicyProto AudioCaptureAllowed = 162;
  optional StringListPolicyProto AudioCaptureAllowedUrls = 210;
  optional BooleanPolicyProto VideoCaptureAllowed = 169;
  optional StringListPolicyProto VideoCaptureAllowedUrls = 211;
  optional BooleanPolicyProto DisableScreenshots = 155;
  optional BooleanPolicyProto TouchVirtualKeyboardEnabled = 271;
  optional BooleanPolicyProto ShowLogoutButtonInTray = 166;
  optional StringPolicyProto ShelfAutoHideBehavior = 168;
  optional StringPolicyProto UserDisplayName = 171;
  optional IntegerPolicyProto SessionLengthLimit = 172;
  optional BooleanPolicyProto FullscreenAllowed = 242;
  optional IntegerPolicyProto ScreenDimDelayAC = 174;
  optional IntegerPolicyProto ScreenOffDelayAC = 175;
  optional IntegerPolicyProto ScreenLockDelayAC = 176;
  optional IntegerPolicyProto IdleWarningDelayAC = 199;
  optional IntegerPolicyProto IdleDelayAC = 177;
  optional IntegerPolicyProto ScreenDimDelayBattery = 178;
  optional IntegerPolicyProto ScreenOffDelayBattery = 179;
  optional IntegerPolicyProto ScreenLockDelayBattery = 180;
  optional IntegerPolicyProto IdleWarningDelayBattery = 200;
  optional IntegerPolicyProto IdleDelayBattery = 181;
  optional IntegerPolicyProto IdleAction = 182;
  optional IntegerPolicyProto IdleActionAC = 228;
  optional IntegerPolicyProto IdleActionBattery = 224;
  optional IntegerPolicyProto LidCloseAction = 183;
  optional BooleanPolicyProto PowerManagementUsesAudioActivity = 184;
  optional BooleanPolicyProto PowerManagementUsesVideoActivity = 185;
  optional IntegerPolicyProto PresentationScreenDimDelayScale = 222;
  optional BooleanPolicyProto AllowScreenWakeLocks = 205;
  optional IntegerPolicyProto UserActivityScreenDimDelayScale = 212;
  optional BooleanPolicyProto WaitForInitialUserActivity = 249;
  optional StringPolicyProto PowerManagementIdleSettings = 260;
  optional StringPolicyProto ScreenLockDelays = 261;
  optional StringPolicyProto TermsOfServiceURL = 188;
  optional BooleanPolicyProto ShowAccessibilityOptionsInSystemTrayMenu = 190;
  optional BooleanPolicyProto LargeCursorEnabled = 213;
  optional BooleanPolicyProto SpokenFeedbackEnabled = 214;
  optional BooleanPolicyProto HighContrastEnabled = 215;
  optional BooleanPolicyProto VirtualKeyboardEnabled = 257;
  optional BooleanPolicyProto KeyboardDefaultToFunctionKeys = 262;
  optional IntegerPolicyProto ScreenMagnifierType = 216;
  optional BooleanPolicyProto AttestationEnabledForUser = 202;
  optional StringListPolicyProto AttestationExtensionWhitelist = 203;
  optional IntegerPolicyProto ContentPackDefaultFilteringBehavior = 206;
  optional StringPolicyProto ContentPackManualBehaviorHosts = 207;
  optional StringPolicyProto ContentPackManualBehaviorURLs = 208;
  optional StringPolicyProto ManagedBookmarks = 229;
  optional StringPolicyProto UserAvatarImage = 251;
  optional StringPolicyProto WallpaperImage = 264;
  optional StringListPolicyProto EnableDeprecatedWebPlatformFeatures = 272;
  optional BooleanPolicyProto EasyUnlockAllowed = 274;
  optional StringListPolicyProto SessionLocales = 276;
  optional StringListPolicyProto CertificateTransparencyEnforcementDisabledForUrls = 337;
  optional BooleanPolicyProto DHEEnabled = 336;
  optional BooleanPolicyProto ForceMaximizeOnFirstRun = 300;
  optional BooleanPolicyProto SSLErrorOverrideAllowed = 302;
  optional BooleanPolicyProto QuicAllowed = 303;
  optional StringPolicyProto KeyPermissions = 304;
  optional BooleanPolicyProto UnifiedDesktopEnabledByDefault = 309;
  optional BooleanPolicyProto ArcEnabled = 319;
  optional StringPolicyProto ArcPolicy = 320;
  optional BooleanPolicyProto SuppressUnsupportedOSWarning = 326;
  optional BooleanPolicyProto TaskManagerEndProcessEnabled = 329;
  optional BooleanPolicyProto AllowScreenLock = 331;
  optional IntegerPolicyProto ArcCertificatesSyncMode = 332;
  optional StringPolicyProto AllowedDomainsForApps = 333;
  optional BooleanPolicyProto PacHttpsUrlStrippingEnabled = 334;
  optional BooleanPolicyProto EnableMediaRouter = 335;
  optional BooleanPolicyProto ArcBackupRestoreEnabled = 339;
  optional StringPolicyProto WebRtcUdpPortRange = 341;
  optional BooleanPolicyProto ComponentUpdatesEnabled = 344;
}

